### DISCUSSION TITLE
Misleading way of not supporting older OS versions

### DISCUSSION THREAD

**Discussion by yesyves** [Homebrew/discussions#467] [Everyday usage]
I have been struggling with the intention of running Homebrew on 10.11 recently and while I have absolutely nothing against the fact that the developers do not want to support it, I find that Brew rather seems to be actively preventing its usage on older versions.

The first problem I encountered was a message about not having Git in my PATH, which I had. After some investigation I found out that I was rather having an outdated version in `/usr/bin` so I installed the 2.29.2 version but still had that message about not having Git in my PATH. So I put that current Git in `/usr/local/bin` and still got the same message that Git was not in my PATH. So I replaced the binary in `/Library/Developer/CommandLineTools/usr/bin` with a current Git binary which was fine with `brew doctor` but I still got the message that Git is not in my PATH with `brew update` ! Which means that the installation process still can't complete. 

The next step would probably be to patch brew's git shim so that 
1. The error message be more specific
but it should rather
2. do what it basically says and actually look for a current enough Git in PATH

The same logic should be applied for curl since 10.11 certificates are outdated.

Since I'm evidently new to Homebrew, being still short of a complete installation, I may have mixed a few things up so any precision or correction will be appreciated.

Thanks !

**Comment by SMillerDev** [comment]
> I find that Brew rather seems to be actively preventing its usage on older versions.

Brew just has a very fast release cycle and we regularly get rid of code that isn't used any more on any supported platform to make sure that the code remains maintainable. If you want to run a 5 year old OS we won't stop you, but it hasn't been a consideration in updates for roughly 3 years so most code supporting it has probably been removed again.
